### PR TITLE
XW-613 | Download scripts in parallel

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/XW-613

Loading script is copied from http://www.html5rocks.com/en/tutorials/speed/script-loading/#toc-dom-rescue

### CODE REVIEW

File: front/scripts/baseline/mercury/utils/loadScripts.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+if (typeof window.M === 'undefined') {
```
Comment (bkoval) [inline front/scripts/baseline/mercury/utils/loadScripts.js]: What if M is null / 0 / '0' / false? Why not if (!window.M) ? :)

File: front/scripts/baseline/mercury/utils/loadScripts.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	 *
+	 * @param {string[]} scripts
+	 * @returns {void}
+	 */
+	M.loadScripts = function (scripts) {
+		let src = scripts.shift(),
+			scriptElement;
+
+		// loop through our script urls
+		while (src) {
```
Comment (hakubo) [inline front/scripts/baseline/mercury/utils/loadScripts.js]: I think for each would be clearer here. What is the benefit of while in this case?

File: front/scripts/baseline/mercury/utils/loadScripts.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+				pendingScripts.push(scriptElement);
+				// listen for state changes
+				scriptElement.onreadystatechange = stateChange;
+				// must set src AFTER adding onreadystatechange listener
+				// else we’ll miss the loaded event for cached scripts
+				scriptElement.src = src;
+			} else {
+				// fall back to defer
+				/* eslint no-useless-concat:0 */
+				document.write(`<script src="${src}" defer><` + `/script>`);
```
Comment (hakubo) [inline front/scripts/baseline/mercury/utils/loadScripts.js]: What is this concatenation for?

Comment (rogatty) [inline front/scripts/baseline/mercury/utils/loadScripts.js]: http://stackoverflow.com/questions/27517750/uncaught-syntaxerror-unexpected-token-illegal-script-in-script-scrip

File: front/scripts/baseline/mercury/utils/loadScripts.js (hunk 1 above)
Comment (rogatty) [inline front/scripts/baseline/mercury/utils/loadScripts.js]: I'm not sure why, just copied it from other utils... @vforge? @bognix?
It's a temporary solution anyway, we plan to rethink the baseline scripts when moving towards Ember CLI.

**Comment by hakubo** [comment]
LGTM

**Comment by tomasznapieralski** [comment]
LGTM

File: front/scripts/baseline/mercury/utils/loadScripts.js (hunk 1 above)
Comment (rogatty) [inline front/scripts/baseline/mercury/utils/loadScripts.js]: I've created a ticket to clean this up https://wikia-inc.atlassian.net/browse/XW-820

Comment (vforge) [inline front/scripts/baseline/mercury/utils/loadScripts.js]: @bkoval Why it should be? Alternatively we can check if it's `=== "object"`

Comment (bkoval) [inline front/scripts/baseline/mercury/utils/loadScripts.js]: Just wondering. Thing is, you want an object here, so the most natural check is to find out if it's not an object. Third party libs or change in how you initialize window.M might expose this line as not foolproof.

!== "object" would be a much safer check, +1 for that.

**Pull request merged**